bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/template/TemplateManager.java

```diff
@@ -0,0 +1,118 @@
+/*******************************************************************************
+ * Copyright (c) 1997, 2015 by ProSyst Software GmbH
+ * http://www.prosyst.com
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    ProSyst Software GmbH - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.smarthome.automation.core.template;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Locale;
+
+import org.eclipse.smarthome.automation.Action;
+import org.eclipse.smarthome.automation.Condition;
+import org.eclipse.smarthome.automation.Trigger;
+import org.eclipse.smarthome.automation.handler.provider.TemplateProvider;
+import org.eclipse.smarthome.automation.template.RuleTemplate;
+import org.eclipse.smarthome.automation.template.Template;
+import org.osgi.framework.BundleContext;
+import org.osgi.util.tracker.ServiceTracker;
+
+/**
+ * @author Yordan Mihaylov - Initial Contribution
+ */
+public class TemplateManager {
```

Comment:
Yes it is required, because our TemplateAdmin interface is different from your TempleteRegistry interface and the TempleteManager contains common implementation for both interfaces. In this way the common logic is the same on both places and  only the implementation of interfaces are changed by the refactor tool.